library/core/src/mem/mod.rs

```diff
@@ -580,6 +580,21 @@ pub const fn needs_drop<T>() -> bool {
     intrinsics::needs_drop::<T>()
 }
 
+/// Returns `true` if finalizing values of type `T` inside GC matters.
+///
+/// This is purely an optimization hint, and may be implemented conservatively:
+/// it may return `true` for types that don't actually need to be finalized.
+/// As such always returning `true` would be a valid implementation of
+/// this function. However if this function actually returns `false`, then you
+/// can be certain omitting a finalizer for `T` has no side effect.
```

Comment:
"certain THAT"

Reply:
Fixed: 06ee0e6f8d034786d5145c1f35c49a06e5e3a903